### DISCUSSION TITLE
Linking to a materials tuning demo

### DISCUSSION THREAD

**Discussion by stefanbringuier** [sparks-baird/self-driving-lab-demo#71] [Demo extensions and design]
@sgbaird awesome awesome project, my plan is to work through this demo in the next couple months! One question is do you think there is a low cost way to do something like automate, using [syringe pumps](https://www.appropedia.org/Open-source_syringe_pump) [^a], the color tuning of [clear 2-part epoxy](https://www.amazon.com/Gorilla-Epoxy-Minute-ounce-Syringe/dp/B01M7VD07W/ref=sr_1_3?keywords=2+part+epoxy&qid=1663372219&sr=8-3)  using  droppers of  red, green, and blue food coloring? The idea is to use the same setup but compare target absorption profile to that measured to find the optimal "processing" steps. Then there is the need to create some kind of track system to move the substrate and dispense a new sample. Well just wanted to get your thoughts and maybe brainstorm.

[^a]: This DIY seems low-cost but requires a 3D printer (:moneybag: )

**Comment by sgbaird** [comment]
> @sgbaird awesome awesome project, my plan is to work through this demo in the next couple months!

Thank you! That's great to hear. I'm posting a final-ish version of the build instructions soon and will let you know when those are out.

> One question is do you think there is a low cost way to do something like automate, using [syringe pumps](https://www.appropedia.org/Open-source_syringe_pump) [1](#user-content-fn-a-35706a3f8d7408de8efd9293cb541ade), the color tuning of [clear 2-part epoxy](https://www.amazon.com/Gorilla-Epoxy-Minute-ounce-Syringe/dp/B01M7VD07W/ref=sr_1_3?keywords=2+part+epoxy&qid=1663372219&sr=8-3) using droppers of red, green, and blue food coloring? The idea is to use the same setup but compare target absorption profile to that measured to find the optimal "processing" steps.

I think so! There's usually that trade-off between cost and setup time. Without a 3D printer, ordering 3D printed parts as a service is also an option. Could you describe the motivation of the idea in more detail?

I think GOSH had some mention of an open source syringe pump as well.

> Then there is the need to create some kind of track system to move the substrate and dispense a new sample. Well just wanted to get your thoughts and maybe brainstorm.

Something from HiWonder might help here. I remember seeing a track system that seemed appealing.

I have some more thoughts and will plan on circling back to this tomorrow or early next week for some more in-depth comments.

**Comment by stefanbringuier** [comment]
> ... ordering 3D printed parts as a service is also an option.

Hmm, didn't think about this. Good point!

> Could you describe the motivation of the idea in more detail?

So I don't have a strong materials engineering motivation here, but what I'm wanting to explore is getting my "hands-dirty" in automated materials processing-property search. I figured I'd build off your project here so I could explore a simple demo to automate tuning of visible spectrum filter. Once I work through this, then I'll start adapting aspects into my research, although I see challenges ahead given the projects are mostly going to be multiobjective optimization problem; hopefully projects like [Chimera](https://github.com/aspuru-guzik-group/chimera) can assist.

> ... HiWonder might help here. I remember seeing a track system that seemed appealing.

Thanks, I'll definitely search around on HiWonder. To be honest, I was thinking maybe a [lego conveyor belt setup](https://frugalfun4boys.com/build-working-lego-conveyor-belt/) (lol) and replacing the belt with clear thin PVC; I haven't thought this out that thoroughly!

**Reply by sgbaird** [reply]
> > ... ordering 3D printed parts as a service is also an option.
> 
> Hmm, didn't think about this. Good point!

I took a look at the open source syringe you mentioned, and it looks like it would be really amenable to ordering 3D-printed parts as a service given the low number of parts and volume. There might also be some local venues that offer this service too (for example, local universities). If you do find a service that you like, would love to hear about it.

> > Could you describe the motivation of the idea in more detail?
> 
> So I don't have a strong materials engineering motivation here, but what I'm wanting to explore is getting my "hands-dirty" in automated materials processing-property search. I figured I'd build off your project here so I could explore a simple demo to automate tuning of visible spectrum filter. Once I work through this, then I'll start adapting aspects into my research, although I see challenges ahead given the projects are mostly going to be multiobjective optimization problem; hopefully projects like [Chimera](https://github.com/aspuru-guzik-group/chimera) can assist.

It sounds like you landed on this project for exactly the reason it was intended for which is great to hear. I'm planning on making a notebook example of multi-objective optimization using the individual wavelengths as objectives instead of a combined error metric, so I think that could be of help at least in terms of the framework. Personally, I think using a noise-tolerant expected hypervolume improvement is the way to go with multi-objective optimization, although I'm pretty interested in seeing how Chimera compares to this. I've been impressed as I've learned more about how Chimera operates.

> > ... HiWonder might help here. I remember seeing a track system that seemed appealing.
> 
> Thanks, I'll definitely search around on HiWonder. To be honest, I was thinking maybe a [lego conveyor belt setup](https://frugalfun4boys.com/build-working-lego-conveyor-belt/) (lol) and replacing the belt with clear thin PVC; I haven't thought this out that thoroughly!

Hmm, I checked again, and the ones I was thinking of are probably too pricey for this project https://hiwonder.hk/collections/conveyor-belt-sliding-rail.

Another option would be a Geneva wheel with a motor. Limits the number of samples, but you could also get creative and extend the syringe out radially. Essentially, it gives you XYZ positioning of sorts with 2 axes that are standalone from each other. Maybe there are better options. DIY conveyer belt could give you "unlimited" sample positions with only a single axis, which certainly has its appeal.

Overall sounds like a really nice project! I'd love to see this as an extension to SDL-Demo. You might also consider taking a look at https://github.com/CytronTechnologies/MAKER-PI-PICO/issues/13 which has some resources related to controlling motors in the context of SDL-Demo.

Aside: here's [a link to an unlisted YouTube video](https://www.youtube.com/watch?v=GVdfJCsQ8vk&t=1s) that shows detailed build instructions for SDL-Demo. Happy to hear feedback!

**Reply by sgbaird** [reply]
@stefanbringuier, here is the multi-objective optimization notebook https://github.com/sparks-baird/self-driving-lab-demo/blob/main/notebooks/6.1-multi-objective.ipynb

**Reply by stefanbringuier** [reply]
@sgbaird awesome! Thanks for sharing, will look through it. I've been enjoying working through this lab demo project.
